gui/wxpython/startup/locdownload.py

```diff
@@ -95,20 +97,62 @@ class DownloadError(Exception):
     """Error happened during download or when processing the file"""
     pass
 
+
 class RedirectText(object):
     def __init__(self, window):
         self.out = window
 
     def write(self, string):
         try:
-            wx.CallAfter(self.out.SetLabel, string)
+            if self.out:
+                string = self._wrap_string(string)
+                heigth = self._get_heigth(string)
+                wx.CallAfter(self.out.SetLabel, string)
+                self._resize(heigth)
         except:
             # window closed -> PyDeadObjectError
             pass
 
     def flush(self):
         pass
 
+    def _wrap_string(self, string, width=40):
+        """Wrap string
+
+        :param str string: input string
+        :param int width: maximum length allowed of the wrapped lines
+
+        :return str: newline-separated string
+        """
+        wrapper = textwrap.TextWrapper(width=width)
+        return wrapper.fill(text=string)
+
+    def _get_heigth(self, string):
+        """Get widget new heigth
+
+        :param str string: input string
+
+        :return int: widget heigth
+        """
+        n_lines = string.count('\n')
+        attr = self.out.GetClassDefaultAttributes()
+        font_size = attr.font.GetPointSize()
+        heigth = int((n_lines + 2) * font_size // 0.75) # 1 px = 0.75 pt
+        return heigth
+
+    def _resize(self, heigth=-1):
+        """Resize widget heigth
+
+        :param int heigth: widget heigth
+        """
+        wx.CallAfter(self.out.GetParent().SetMinSize, (-1, -1))
+        wx.CallAfter(self.out.SetMinSize, (-1, heigth))
+        wx.CallAfter(
+            self.out.GetParent().parent.sizer.Fit,
+            self.out.GetParent().parent,
+        )
+
+
```

Comment:
Is this all really needed? I don't have time to look at it now, but I am sure there must be a more elegant solution.

Reply:
> Is this all really needed? I don't have time to look at it now, but I am sure there must be a more elegant solution.

Yes it is necessary. The original code didn't address it at all. There is a `wx.StaticText` widget `Wrap()` method, but it has a problem with e.g. long url string that does not break. It is also necessary to adjust the height of the message `wx.StaticText` widget dynamically depending on the length of the text. In a short time I did not find a better solution.